From 70362cee62513e88a215000ebf13a2f1c14adf73 Mon Sep 17 00:00:00 2001
From: Alan Tull <atull@opensource.altera.com>
Date: Mon, 23 May 2016 15:08:58 -0500
Subject: [PATCH 2/2] arria10 Device Tree overlays for sysid test

Adds 3 overlays, one for the static FPGA image plus two overlay
personas.
---
 arch/arm/boot/dts/Makefile                         |   3 +
 arch/arm/boot/dts/persona0.dts                     |  34 +++++++
 arch/arm/boot/dts/persona1.dts                     |  34 +++++++
 ...ocfpga_arria10_socdk_sdmmc_ghrd_ovl_ext_cfg.dts | 107 +++++++++++++++++++++
 4 files changed, 178 insertions(+)
 create mode 100644 arch/arm/boot/dts/persona0.dts
 create mode 100644 arch/arm/boot/dts/persona1.dts
 create mode 100644 arch/arm/boot/dts/socfpga_arria10_socdk_sdmmc_ghrd_ovl_ext_cfg.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 8382337..97e385b9 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -877,6 +877,9 @@ dtb-$(CONFIG_ARCH_SOCFPGA) += \
 	socfpga_cyclone5_sodia.dtb \
 	socfpga_cyclone5_trcom.dtb \
 	socfpga_cyclone5_vining_fpga.dtb \
+	persona0.dtb \
+	persona1.dtb \
+	socfpga_arria10_socdk_sdmmc_ghrd_ovl_ext_cfg.dtb \
 	socfpga_vt.dtb
 dtb-$(CONFIG_ARCH_SPEAR13XX) += \
 	spear1310-evb.dtb \
diff --git a/arch/arm/boot/dts/persona0.dts b/arch/arm/boot/dts/persona0.dts
new file mode 100644
index 0000000..1bc171e
--- /dev/null
+++ b/arch/arm/boot/dts/persona0.dts
@@ -0,0 +1,34 @@
+/dts-v1/;
+/plugin/;
+/ {
+	fragment@1 {
+		target-path = "/soc/base_fpga_region/fpga_pr_region0";
+		#address-cells = <1>;
+		#size-cells = <1>;
+		__overlay__ {
+			partial-fpga-config;
+			firmware-name = "persona0.rbf";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			region-unfreeze-timeout-us = <4>;
+			region-freeze-timeout-us = <4>;
+			config-complete-timeout-us = <100>;
+
+			pr_region_0_pr_clk_100: pr_region_0_pr_clk_100 {
+				compatible = "fixed-clock";
+				#clock-cells = <0>;
+				clock-frequency = <100000000>;	/* 100.00 MHz */
+				clock-output-names = "pr_region_0_pr_clk_100-clk";
+			};
+
+			pr_region_0_pr_sysid_qsys_0: sysid@0x0800 {
+				compatible = "altr,sysid-16.1", "altr,sysid-1.0";
+				reg = <0x00000800 0x00000008>;
+				clocks = <&pr_region_0_pr_clk_100>;
+				id = <3405707982>;	/* embeddedsw.dts.params.id type NUMBER */
+				timestamp = <0>;	/* embeddedsw.dts.params.timestamp type NUMBER */
+			};
+		};
+	};
+};
diff --git a/arch/arm/boot/dts/persona1.dts b/arch/arm/boot/dts/persona1.dts
new file mode 100644
index 0000000..6de155a
--- /dev/null
+++ b/arch/arm/boot/dts/persona1.dts
@@ -0,0 +1,34 @@
+/dts-v1/;
+/plugin/;
+/ {
+	fragment@1 {
+		target-path = "/soc/base_fpga_region/fpga_pr_region0";
+		#address-cells = <1>;
+		#size-cells = <1>;
+		__overlay__ {
+			partial-fpga-config;
+			firmware-name = "persona1.rbf";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			region-unfreeze-timeout-us = <4>;
+			region-freeze-timeout-us = <4>;
+			config-complete-timeout-us = <100>;
+
+			pr_region_0_pr_clk_100: pr_region_0_pr_clk_100 {
+				compatible = "fixed-clock";
+				#clock-cells = <0>;
+				clock-frequency = <100000000>;	/* 100.00 MHz */
+				clock-output-names = "pr_region_0_pr_clk_100-clk";
+			};
+
+			pr_region_0_pr_sysid_qsys_0: sysid@0x0900 {
+				compatible = "altr,sysid-16.1", "altr,sysid-1.0";
+				reg = <0x00000900 0x00000008>;
+				clocks = <&pr_region_0_pr_clk_100>;
+				id = <3405707982>;	/* embeddedsw.dts.params.id type NUMBER */
+				timestamp = <0>;	/* embeddedsw.dts.params.timestamp type NUMBER */
+			};
+		};
+	};
+};
diff --git a/arch/arm/boot/dts/socfpga_arria10_socdk_sdmmc_ghrd_ovl_ext_cfg.dts b/arch/arm/boot/dts/socfpga_arria10_socdk_sdmmc_ghrd_ovl_ext_cfg.dts
new file mode 100644
index 0000000..d6458dc
--- /dev/null
+++ b/arch/arm/boot/dts/socfpga_arria10_socdk_sdmmc_ghrd_ovl_ext_cfg.dts
@@ -0,0 +1,107 @@
+/dts-v1/;
+/plugin/;
+/ {
+	fragment@0 {
+		target-path = "/soc/base_fpga_region";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		__overlay__ {
+			ranges = <0xc0000000 0xc0000000 0x00040000>,
+				 <0x00000000 0xff200000 0x00001000>;
+
+			external-fpga-config;
+
+			clocks = <&clk_0 &clk_0 &clk_0 &clk_0>;
+			clock-names = "h2f_axi_clock", "h2f_lw_axi_clock", "f2sdram0_clock", "f2sdram2_clock";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			clk_0: clk_0 {
+				compatible = "fixed-clock";
+				#clock-cells = <0>;
+				clock-frequency = <100000000>;	/* 100.00 MHz */
+				clock-output-names = "clk_0-clk";
+			};
+
+			fpga_pr_region0 {
+				compatible = "fpga-region";
+				fpga-bridges = <&freeze_controller_0>;
+				ranges;
+				#address-cells = <1>;
+				#size-cells = <1>;
+			};
+
+			ILC: ilc@0x0100 {
+				compatible = "altr,altera_ilc-16.1", "altr,ilc-1.0";
+				reg = <0x00000100 0x00000100>;
+				interrupt-parent = <&intc>;
+				interrupts = <0 21 4 0 19 1 0 20 1>;
+				interrupt-names = "freeze_controller_0", "button_pio", "dipsw_pio";
+				interrupt-controller;
+				#interrupt-cells = <1>;
+				clocks = <&clk_0>;
+				altr,sw-fifo-depth = <32>;
+				status = "disabled";
+			};
+
+			freeze_controller_0: freeze_controller@0x0450 {
+				compatible = "altr,freeze_controller-16.1", "altr,freeze-bridge-controller";
+				reg = <0x00000450 0x00000010>;
+				interrupt-parent = <&intc>;
+				interrupts = <0 21 4>;
+				clocks = <&clk_0>;
+			};
+
+			sysid_qsys_0: sysid@0x0000 {
+				compatible = "altr,sysid-16.1", "altr,sysid-1.0";
+				reg = <0x00000000 0x00000008>;
+				clocks = <&clk_0>;
+				id = <3221755904>;
+				timestamp = <0>;
+			};
+
+			led_pio: gpio@0x0010 {
+				compatible = "altr,pio-16.1", "altr,pio-1.0";
+				reg = <0x00000010 0x00000010>;
+				clocks = <&clk_0>;
+				altr,gpio-bank-width = <4>;
+				resetvalue = <0>;
+				#gpio-cells = <2>;
+				gpio-controller;
+			};
+
+			button_pio: gpio@0x0020 {
+				compatible = "altr,pio-16.1", "altr,pio-1.0";
+				reg = <0x00000020 0x00000010>;
+				interrupt-parent = <&intc>;
+				interrupts = <0 19 1>;
+				clocks = <&clk_0>;
+				altr,gpio-bank-width = <4>;
+				altr,interrupt-type = <2>;
+				altr,interrupt_type = <2>;
+				edge_type = <1>;
+				level_trigger = <0>;
+				resetvalue = <0>;
+				#gpio-cells = <2>;
+				gpio-controller;
+			};
+
+			dipsw_pio: gpio@0x0030 {
+				compatible = "altr,pio-16.1", "altr,pio-1.0";
+				reg = <0x00000030 0x00000010>;
+				interrupt-parent = <&intc>;
+				interrupts = <0 20 1>;
+				clocks = <&clk_0>;
+				altr,gpio-bank-width = <4>;
+				altr,interrupt-type = <3>;
+				altr,interrupt_type = <3>;
+				edge_type = <2>;
+				level_trigger = <0>;
+				resetvalue = <0>;
+				#gpio-cells = <2>;
+				gpio-controller;
+			};
+		};
+	};
+};
-- 
2.7.4

